Automattic/newspack-blocks
# CI: configure semantic release for automated releases

### All Submissions:






### Changes proposed in this Pull Request:



This configures commit linting and semantic-release:
**Commit linting** simply enforces a [structure of a commit message](https://semantic-release.gitbook.io/semantic-release/#commit-message-format), in this case to adhere to [Angular Commit Message Conventions](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commit-message-format)
**semantic-release** is configured to run when anything lands on `release` branch. In such event, the following should happen (if there are any commits that should trigger a release!):
1. a new version number will be determined
1. a zipped release archive  will be created (`npm run release:archive`)
1. `CHANGELOG.md` file will be updated (or generated)
1. version in `package.json`, `package-lock.json`, and `newspack-blocks.php` will be updated
1. the above will be committed by a bot to the repo
1. the bot will also publish a new release on Github, with the zipped archive and release notes

### How to test the changes in this Pull Request:

To test commit message linting:

1. Update git hooks by running `./vendor/brainmaestro/composer-git-hooks/cghooks update`
1. Add some changes to git staging
1. Try to commit with an invalid commit message (e.g. `git commit -m "hehe"`)
1. Observe that the commitlint won't have any of it
1. Commit with a valid commit message (e.g. `git commit -m "feat: adjust header size"`)
1. Observe that the commit was successful

### Other information:
